### PR TITLE
support https in the proxy

### PR DESCRIPTION
**Pull request by yesil**
This PR introduces mas proxy for Odin to use on the hlx domains.

It is already configured and functional.
Resolve: https://jira.corp.adobe.com/browse/MWPW-160093

Test URLs:
- Before: https://main--mas--adobecom.hlx.live/studio.html
- After: https://mwpw-160093--mas--adobecom.hlx.live/studio.html#query=acom

### CODE REVIEW

**Review by 3ch023 — COMMENTED** [review]
@yesil could you add details where the proxy is configured to https://wiki.corp.adobe.com/display/adobedotcom/M@S2+Developer+Onboarding?

Meaning if tomorrow I want to switch https://mas-proxy-odin.corp.adobe.com to point to Odin Stage, how do I do that?

**Comment by yesil** [comment]
@3ch023 it is already in the doc: https://github.com/adobecom/mas/tree/main/studio#running-the-proxy

**Review by 3ch023 — APPROVED** [review]

File: studio/proxy-server.mjs [hunk 1]
```diff
@@ -11,25 +13,40 @@ const corsHeaders = {
...
+    // Read the SSL certificate and private key
+    const httpsOptions = {
+        key: readFileSync(keyPath),
+        cert: readFileSync(certPath),
+    };
+
+    server = createHttpsServer(httpsOptions, requestHandler);
+    serverProtocol = 'https';
+    serverPort = 8443; // Use 8443 for HTTPS
+} else {
```
Comment (npeltier) [inline studio/proxy-server.mjs:41]: why leaving http option? i would remove it tbh

**Review by npeltier — APPROVED** [review]
thanks! hope it won't be useful very long :)

File: studio/proxy-server.mjs (hunk 1 above)
Comment (yesil) [reply studio/proxy-server.mjs:41]: it is needed on local. If we drop local proxy support we can remove it.

**Pull request merged**